File: backend/src/main/java/com/project/backend/domain/follow/exception/FollowException.java
package com.project.backend.domain.follow.exception;

import org.springframework.http.HttpStatus;

public class FollowException extends RuntimeException {
    private final FollowErrorCode followErrorCode;

    public FollowException(FollowErrorCode followErrorCode) {
        super(followErrorCode.message);
        this.followErrorCode = followErrorCode;
    }

    public HttpStatus getStatus() {
        return followErrorCode.status;
    }

    public String getCode() {
        return followErrorCode.code;
    }
}

File: backend/src/main/java/com/project/backend/domain/follow/exception/FollowException.kt
package com.project.backend.domain.follow.exception;

import org.springframework.http.HttpStatus;

class FollowException(
    private val followErrorCode: FollowErrorCode
) : RuntimeException(followErrorCode.message) {

    fun getStatus(): HttpStatus {
        return followErrorCode.status
    }

    fun getCode(): String {
        return followErrorCode.code
    }
}
